packages/core/src/pages/[...slug].tsx

```diff
@@ -138,6 +143,15 @@ export const getStaticProps: GetStaticProps<
     globalSectionsPromise,
   ])
 
+  const [serverManyProductsData, serverManyProductsVariables] =
+    await fetchServerManyProducts({
+      itemsPerPage: cmsPage?.settings?.productGallery?.itemsPerPage,
+      sort: cmsPage?.settings?.productGallery
+        ?.sortBySelection as SearchState['sort'],
+      term: '',
+      selectedFacets: data?.collection?.meta.selectedFacets,
+    })
```

Comment:
fetch products from SSR, also uses serverManyProductsVariables for cache mechanism.